src/parse.rs

```diff
@@ -127,21 +144,23 @@ fn block_comment(input: Cursor) -> PResult<&str> {
 
     let mut depth = 0;
     let bytes = input.as_bytes();
-    let mut i = 0;
-    let upper = bytes.len() - 1;
-    while i < upper {
+    let mut iter = input.char_offsets();
+
+    while let Some((ch_i, i, _)) = iter.next() {
         if bytes[i] == b'/' && bytes[i + 1] == b'*' {
```

Comment:
This panicks now because the condition `i < upper` is removed. For example see `"/*/".parse::<proc_macro2::TokenStream>()`.

Reply:
:facepalm: